File: binary_search/binary_search.c
#include <stdio.h>

int binary_search(int arr[], int n, int k){
    int l = 0;
    int r = n-1;
    while(l<=r){
        int m = (int)(l+r)/2;
        if(arr[m] == k) return m;
        else if(k < arr[m]) r = m-1;
        else l = m+1;
    }
    return -1;
}

File: binary_search/binary_search.py
def binary_search(arr, k):
    l = 0
    r = len(arr)-1
    while l <=r:
        m = int((l+r)/2)
        if arr[m] == k: return m
        elif k < arr[m]: r = m-1
        else: l = m+1
    return -1
